src/Core/src/Handlers/View/ViewHandler.cs

```diff
@@ -201,10 +227,17 @@ public static void MapIsEnabled(IViewHandler handler, IView view)
 
 		public static void MapVisibility(IViewHandler handler, IView view)
 		{
+			handler.Invoke(nameof(INeedsContainerViewHandler.NeedsContainer), nameof(IView.Visibility));
+
 			if (handler.HasContainer)
+			{
 				((PlatformView?)handler.ContainerView)?.UpdateVisibility(view);
+				((PlatformView?)handler.PlatformView)?.UpdateVisibility(Visibility.Visible);
+			}
 			else
+			{
 				((PlatformView?)handler.PlatformView)?.UpdateVisibility(view);
+			}
 		}
```

Comment:
When a property changes that needs to respond to ContainerView changes, we call the "special" invoke and pass the property that requires a re-fire. This updates a collection, and ALSO triggers the path to ensure the container is correct.

Then based on the HasContainer, the mapper needs to set the values of the container and/or the platform view.